src/RelayProvider.ts

```diff
@@ -0,0 +1,270 @@
+        }
+        if (!requestConfig) {
+            return false;
+        }
+
+        return requestConfig.useEnveloping ?? false;
+    }
+
+    private _getRelayStatus(respResult: TransactionReceipt): {
+        relayRevertedOnRecipient: boolean;
+        transactionRelayed: boolean;
+        reason?: string;
+    } {
+        if (!respResult.logs || respResult.logs.length === 0) {
+            return {
+                relayRevertedOnRecipient: false,
+                transactionRelayed: false,
+                reason: 'Tx logs not found'
+            };
+        }
+
+        const relayHubInterface = RelayHub__factory.createInterface();
+        const parsedLogs = respResult.logs.map(log => relayHubInterface.parseLog(log));
+
+        const recipientRejectedEvents = parsedLogs.find(log => log.name == 'TransactionRelayedButRevertedByRecipient');
+
+        if (recipientRejectedEvents) {
+            const { reason } = recipientRejectedEvents.args;
+
+            const revertReason = (reason as string) ?? 'Unknown';
+            log.info(
+                `Recipient rejected on-chain: ${revertReason}`
+            );
+
+            return {
+                relayRevertedOnRecipient: true,
+                transactionRelayed: false,
+                reason: revertReason
+            };
+        }
+
+        const transactionRelayed = parsedLogs.find(log => log.name == 'TransactionRelayed');
+
+        if (transactionRelayed) {
+            return {
+                relayRevertedOnRecipient: false,
+                transactionRelayed: true
+            };
+        }
+
+        // Check if it wasn't a deploy call which does not emit TransactionRelayed events but Deployed events
+        const deployedEvent = parsedLogs.find(log => log.name == 'Deployed');
+
+        if (deployedEvent) {
+            return {
+                relayRevertedOnRecipient: false,
+                transactionRelayed: true
+            };
+        }
+
+        log.info(
+            'Neither TransactionRelayed, Deployed, nor TransactionRelayedButRevertedByRecipient events found. This might be a non-enveloping transaction '
+        );
+
+        return {
+            relayRevertedOnRecipient: false,
+            transactionRelayed: false,
+            reason: 'Neither TransactionRelayed, Deployed, nor TransactionRelayedButRevertedByRecipient events found. This might be a non-enveloping transaction'
+        };
+    }
+
+    private async executeRelayTransaction(
+        envelopingRequest: UserDefinedEnvelopingRequest,
+        requestConfig: RequestConfig
+    ): Promise<RelayingResult> {
+        try {
+
+            const transaction = await this.relayClient.relayTransaction(
+                envelopingRequest, requestConfig
+            );
+
+            const { hash } = transaction;
+
+            if(!hash){
+                throw new Error('Transaction has no hash!');
+            }
+            const receipt = await this.getTransactionReceipt(hash);
```

Comment:
Does the JsonRpcProvider always wait for the transactionReceipt to be ready? If not, I don't think we should do the same here

Reply:
We could eventually add an optional parameter to let the user decide what to do.

Reply:
Actually, the provider used to have this functionality, I just missed it so I will be adding it back.

Reply:
We reverted this to wait for transaction request only if `ignoreTransactionReceipt` is false